packages/core/src/interpreter.ts

```diff
@@ -432,8 +433,9 @@ export class Interpreter<
   }
 
   private handleError(errorData: unknown): void {
-    const errorInstance =
-      errorData instanceof Error ? errorData : new Error(errorData as any);
+    const errorInstance = isError(errorData)
+      ? errorData
+      : new Error(errorData as any);
```

Comment:
I believe that this still would result in a degraded DX because if we throw non-error object we'll end up with stringified messages like `[object Object]` which ain't that helpful. Whatever gets thrown should just be preserved and we shouldn't try to wrap it in any way - it mostly should be treated as opaque by us.

Reply:
Fixed